python/mlx/nn/layers/dropout.py

```diff
@@ -33,3 +33,45 @@ def __call__(self, x):
         mask = mx.random.bernoulli(self._p_1, x.shape)
 
         return (1 / self._p_1) * mask.astype(x.dtype) * x
+
+
+class Dropout2d(Module):
+    """Randomly zero an entire channel. During training, each channel
+    will be zeroed-out independently with probability p.
+
+    The remaining channels are multiplied with :math:`\frac{1}{1-p}` where
+    :math:`p` is the probability of zeroing an element. This is done so the
+    expected value of a given element will remain the same.
+
+    Args:
+        p (float): The probability to zero a channel
+    """
```

Comment:
- Could you say a bit more about the expected ordering / shape of the input?
- Also for our Conv2D we do `NHWC` by default, so probably we should change this to match that?

Reply:
Yes, my bad, I missed that convolutions used a different convention 🥲 Will update